### PR TITLE
fix for  undefined index errors

### PR DESCRIPTION
**Pull request by afragen**
This fixes a couple of 'undefined index' errors I was seeing.

### CODE REVIEW

File: includes/class-updater.php [hunk 1]
```diff
@@ -96,7 +96,7 @@ public static function updates_this_plugin( $plugin ) {
 
 		$uri = static::parse_plugin_uri( $plugin );
 
-		if ( in_array( $uri['host'], static::$valid_domain_names ) ) {
+		if ( isset( $uri['host'] ) && in_array( $uri['host'], static::$valid_domain_names ) ) {
```
Comment (pdclark) [inline includes/class-updater.php:99]: What value of `Git URI` are you passing in the plugin that `host` isn't coming through? No valid `URI` would be without a host.

File: includes/class-updater.php [hunk 2]
```diff
@@ -239,9 +239,12 @@ protected function set_repo_info( $plugin ) {
 		$uri  = self::parse_plugin_uri( $plugin );
 		$path = explode('/', $uri['path'] );
 
-		$this->host       = $uri['host'];
-		$this->username   = str_replace( '%40', '@', $uri['user'] );
-		$this->password   = $uri['pass'];
+		if ( isset( $uri['user'], $uri['pass'] ) ) {		
+			$this->host       = $uri['host'];
```
Comment (pdclark) [inline includes/class-updater.php]: Host should be above if statement

**Comment by afragen** [comment]
Again, must be a short uri issue.

File: includes/class-updater.php (hunk 2 above)
Comment (afragen) [inline includes/class-updater.php]: I'll fix

Comment (afragen) [inline includes/class-updater.php]: I think if we're going to support short uri that we should leave it where it is. The other choice is to parse the actual headers like in `github-updater`.

**Comment by pdclark** [comment]
I'm fairly against short URI by default. It's an obscure usage, and directly contradicts what a `URI` _is_ — it's by definition a full address including host. If Gary or others want to use it for individual instances, it can either be provided by a filter, or a filter can be provided to disable updates from the updater (which makes more sense in my mind — I don't think anyone would see the connection between the short vs long "URI" and updating or not).

File: includes/class-updater.php (hunk 2 above)
Comment (pdclark) [inline includes/class-updater.php]: Left as-is, it's still not correct. Setting host needs to be based on a check for host being set, not username and password— the presence of the two are not related.

Comment (afragen) [inline includes/class-updater.php]: I'll fix it.

**Pull request merged**